asyncapi-cli/src/main/java/io/ballerina/asyncapi/codegenerator/controller/DispatcherController.java

```diff
@@ -99,6 +152,57 @@ private FunctionDefinitionNode getResourceFuncNode(ModulePartNode oldRoot) {
         }
         return null;
     }
+
+    private FunctionDefinitionNode getPostFuncNode(ModulePartNode oldRoot) {
+        for (ModuleMemberDeclarationNode node : oldRoot.members()) {
+            if (node.kind() == SyntaxKind.CLASS_DEFINITION) {
+                for (Node funcNode : ((ClassDefinitionNode) node).members()) {
+                    if ((funcNode.kind() == SyntaxKind.RESOURCE_ACCESSOR_DEFINITION)
+                            && ((FunctionDefinitionNode) funcNode).functionName().text().equals(
+                            Constants.DISPATCHER_SERVICE_POST_FUNCTION_NAME)) {
+                        return (FunctionDefinitionNode) funcNode;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    private VariableDeclarationNode getEventIdentifierNode() {
+        // {@code string eventIdentifier}
+        BuiltinSimpleNameReferenceNode typeBindingPattern = createBuiltinSimpleNameReferenceNode(null,
+                createToken(STRING_KEYWORD));
+        CaptureBindingPatternNode bindingPattern = createCaptureBindingPatternNode(
+                createIdentifierToken("eventIdentifier"));
+        TypedBindingPatternNode typedBindingPatternNode = createTypedBindingPatternNode(typeBindingPattern,
+                bindingPattern);
+
+        // {@code check request.getHeader("event-name")}
+        SimpleNameReferenceNode variableName = createSimpleNameReferenceNode(createIdentifierToken("request"));
+        SimpleNameReferenceNode methodName = createSimpleNameReferenceNode(createIdentifierToken("getHeader"));
+        SimpleNameReferenceNode argumentName =
+                createSimpleNameReferenceNode(createIdentifierToken("\"" + this.eventIdentifierPath + "\""));
+        FunctionArgumentNode eventIdentifierArgument = createPositionalArgumentNode(argumentName);
+        SeparatedNodeList<FunctionArgumentNode> arguments = createSeparatedNodeList(eventIdentifierArgument);
+        MethodCallExpressionNode methodCallExpression =
+                createMethodCallExpressionNode(variableName, createToken(DOT_TOKEN), methodName,
+                        createToken(OPEN_PAREN_TOKEN), arguments, createToken(CLOSE_PAREN_TOKEN));
+        CheckExpressionNode initializer =
+                createCheckExpressionNode(null, createToken(CHECK_KEYWORD), methodCallExpression);
+
+        // {@code string eventIdentifier = check request.getHeader("event-name");}
+        VariableDeclarationNode eventIdentifierNode =
+                createVariableDeclarationNode(createEmptyNodeList(), null, typedBindingPatternNode,
+                        createToken(EQUAL_TOKEN), initializer, createToken(SEMICOLON_TOKEN));
+        return eventIdentifierNode;
+    }
+
+    private Boolean isEventIdentifierInHeader(String eventIdentifierPath) {
+        if (eventIdentifierPath.startsWith(Constants.CLONE_WITH_TYPE_VAR_NAME)) {
```

Comment:
Shall we check whether the `type` field of `x-ballerina-event-identifier` is `header` and determine this?

Reply:
This change is done in [b6ae9c1](https://github.com/ballerina-platform/asyncapi-tools/pull/54/commits/b6ae9c1a6a84c715f44126e6cd85ee493177a302)